includes/functions/functions_error_handling.php

```diff
@@ -0,0 +1,160 @@
+}
+
+function zen_fatal_error_handler()
+{
+    $last_error = error_get_last();
+
+    if (!empty($last_error) && in_array($last_error['type'], [E_ERROR, E_USER_ERROR, E_PARSE])) {
+        $message = date('[d-M-Y H:i:s e]') . ' Request URI: ' . ($_SERVER['REQUEST_URI'] ?? 'not set') . ', IP address: ' . ($_SERVER['REMOTE_ADDR'] ?? 'not set') . PHP_EOL;
+        $message_type = ($last_error['type'] == E_PARSE) ? 'Parse' : (($last_error['type'] == E_RECOVERABLE_ERROR) ? 'Catchable Fatal' : 'Fatal');
+        $message .= "--> PHP $message_type error: {$last_error['message']} in {$last_error['file']} on line {$last_error['line']}.";
+        error_log(PHP_EOL . $message . PHP_EOL, 3, $GLOBALS['debug_logfile_path']);
+    }
+}
+
+/**
+ * The path where the debug log file will be located
+ * Default value is: DIR_FS_LOGS . '/myDEBUG-yyyymmdd-hhiiss-xxxxx.log' (storefront) or DIR_FS_LOGS . '/myDEBUG-adm-yyyymmdd-hhiiss-xxxxx.log' (admin)
+ * ... which puts it in the /logs/ folder:   /logs/myDEBUG-yyyymmdd-hhiiss-xxxxx.log or /logs/myDEBUG-yyyymmdd-hhiiss-xxxxx.log, respectively.
+ *     where:
+ *      - yyyy .... is the 4-digit year
+ *      - mm ...... is the 2-digit month
+ *      - dd ..... is the 2-digit day-of-month
+ *      - hh ..... is the 2-digit hour
+ *      - ii ..... is the 2-digit minute
+ *      - ss ..... is the 2-digit second
+ *      - xxxxx ... is the time in milliseconds
+ *
+ *    (or if you don't have a /logs/ folder, it will use the /cache/ folder instead)
+ *
+ * The error-handler now groups the logs by severity, based on the log-file's suffix, examples given for the storefront, by
+ * the zen_debug_error_handler function (above). For the admin, the '-adm' addition described above is included in the files' name(s).
+ *
+ * PHP notices:      /logs/myDEBUG-yyyymmdd-hhiiss-xxxxx.log
+ * PHP deprecations: /logs/myDEBUG-yyyymmdd-hhiiss-xxxxx-deprecated.log
+ * PHP warnings:     /logs/myDEBUG-yyyymmdd-hhiiss-xxxxx-warning.log
+ * PHP errors:       /logs/myDEBUG-yyyymmdd-hhiiss-xxxxx-error.log
+ */
+function zen_set_error_logging_filename(): string
+{
+    /**
+     * Used as a global by registered zen_debug_error_handler and zen_fatal_error_handler functions
+     */
+    global $debug_logfile_path;
+
+    $log_prefix = (IS_ADMIN_FLAG) ? '/myDEBUG-adm-' : '/myDEBUG-';
+    $log_date = new DateTime();
+    $debug_logfile_path = DIR_FS_LOGS . $log_prefix . $log_date->format('Ymd-His-u') . '.log';
+
+    @ini_set('error_log', $debug_logfile_path);  // the filename to log errors into
+
+    return $debug_logfile_path;
+}
+
+function zen_enable_error_logging(array $pages_to_debug = ['*'], $logging_level = E_ALL & ~E_NOTICE): void
+{
+    global $current_page_base, $debug_logfile_path;
+
+    if (empty($debug_logfile_path)) {
+        $debug_logfile_path = zen_set_error_logging_filename();
+    }
+
+    if (in_array('*', $pages_to_debug, true) || in_array($current_page_base, $pages_to_debug, true)) {
```

Comment:
`$pages_to_debug` should be set in global scope.

Reply:
why does it need to be in the global scope?  the var is passed as an input parm to the method.

what am i missing?

Reply:
as a side note, via the terminal, i was able to download this pr (and test) using the following command:

```
git fetch upstream pull/5966/head:error-handling
```
i am sure there is something similar in the git client that you are using.